### PR TITLE
nixosTests.nix-misc: split from nixosTests.misc

### PR DESCRIPTION
**Pull request by Artturin**
Bit weird to have nix tests in misc

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/package-management/lix/common.nix [hunk 1]
```diff
@@ -260,7 +260,7 @@ let
     passthru = {
       inherit aws-sdk-cpp boehmgc;
       tests = {
-        misc = nixosTests.misc.lix.passthru.override { nixPackage = self; };
+        misc = nixosTests.nix-misc.lix.passthru.override { nixPackage = self; };
```
Comment (Artturin) [inline pkgs/tools/package-management/lix/common.nix:263]: @RaitoBezarius This doesn't work before or after this PR or at 3388179c48fa673ae466f12d15cd7864b8e7cd3c when the line was added

```
$ nix build ".#lix.tests.misc"
error: attribute 'passthru' missing
```

The similarly patterned spark works

```
nix-repl> nixosTests.nix-misc.default
«derivation /nix/store/4qvam2nhs4dy5cibswrj1pici8s32h2p-vm-test-run-misc.drv»

nix-repl> nixosTests.spark.default
{ passthru = { ... }; recurseForDerivations = true; sparkCluster = «derivation /nix/store/ym7yki4znrln73ng96rj4fl7rsz95v99-vm-test-run-spark.drv»; }
```

https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/spark/default.nix

Weird...

File: pkgs/tools/package-management/nix/common.nix [hunk 2]
```diff
@@ -251,6 +252,9 @@ self = stdenv.mkDerivation {
 
     tests = {
       nixi686 = pkgsi686Linux.nixVersions.${self_attribute_name};
+      # Basic smoke test that needs to pass when upgrading nix.
+      # Note that this test does only test the nixVersions.stable attribute.
+      misc = nixosTests.nix-misc.default;
```
Comment (Mic92) [inline pkgs/tools/package-management/nix/common.nix:257]: Looks better than my approach, but I believe we also need to update the nixos test in lix?

**Review by Mic92 — APPROVED** [review]
I like this more than my attempt. Haven't tested anything beyond an eye-ball test.

**Comment by vcunat** [comment]
This doesn't add the new test as a channel blocker anywhere.  Might be worth considering.

**Comment by Artturin** [comment]
@RaitoBezarius Is this ok?

**Review by RaitoBezarius — APPROVED** [review]
Looks good to me! Thank you!

**Comment by mweinelt** [comment]
This broke eval, because the misc test wasn't removed from the tested set.

**Comment by Artturin** [comment]
> This broke eval, because the misc test wasn't removed from the tested set.
> 
> ```
> {UNKNOWN}: aggregate job ‘tested’ failed with the error: nixos.tests.misc.default.x86_64-linux: does not exist
> nixos.tests.misc.default.aarch64-linux: does not exist
>  at /nix/store/fncb6n480apg2y1jdzf94ivlw9dd88nr-hydra-perl-deps/lib/perl5/site_perl/5.38.2/Catalyst/Model/DBIC/Schema.pm line 526
> ```

Ah just `.default` suffix has to be removed

**Comment by Mic92** [comment]
@Artturin @mweinelt https://github.com/NixOS/nixpkgs/pull/317172/files

File: pkgs/tools/package-management/lix/common.nix (hunk 1 above)
Comment (RaitoBezarius) [reply pkgs/tools/package-management/lix/common.nix:263]: This is because `discoverTests` will shortcut as soon as it sees the `test` attribute.

**Pull request merged**